packages/react/src/Testimonial/Testimonial.test.tsx

```diff
@@ -0,0 +1,516 @@
+import React from 'react'
+import {render} from '@testing-library/react'
+import '@testing-library/jest-dom'
+import {axe, toHaveNoViolations} from 'jest-axe'
+import {Testimonial} from './Testimonial'
+
+expect.extend(toHaveNoViolations)
+
+const leftDoubleQuote = '“'
+
+describe('Testimonial', () => {
+  const mockQuote = 'GitHub helps us ensure that we have our security controls baked into our pipelines.'
+  const mockName = 'David Ross'
+  const mockPosition = 'Staff Security Engineer'
+  const mockAvatarSrc = '/images/avatar-mona.png'
+  const mockAvatarAlt = 'David Ross avatar'
+
+  const DefaultTestimonial = () => (
+    <Testimonial>
+      <Testimonial.Quote>{mockQuote}</Testimonial.Quote>
+      <Testimonial.Name position={mockPosition}>{mockName}</Testimonial.Name>
+      <Testimonial.Avatar src={mockAvatarSrc} alt={mockAvatarAlt} />
+    </Testimonial>
+  )
+
+  it('has no accessibility violations', async () => {
+    const {container} = render(<DefaultTestimonial />)
+    const results = await axe(container)
+    expect(results).toHaveNoViolations()
+  })
+
+  it('renders with default props', () => {
+    const {getByRole, getByText} = render(<DefaultTestimonial />)
+
+    expect(getByRole('figure')).toBeInTheDocument()
+    expect(getByText(mockQuote)).toBeInTheDocument()
+    expect(getByText(mockName)).toBeInTheDocument()
+    expect(getByText(mockPosition)).toBeInTheDocument()
+    expect(getByRole('img', {name: mockAvatarAlt})).toBeInTheDocument()
+  })
+
+  it('renders quote mark with default color', () => {
+    const {getByText} = render(<DefaultTestimonial />)
+
+    const quoteMark = getByText(leftDoubleQuote)
+    expect(quoteMark).toHaveClass('Testimonial__quoteMark--default')
+  })
+
+  it('applies custom className', () => {
+    const {getByRole} = render(
+      <Testimonial className="custom-testimonial">
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    expect(getByRole('figure')).toHaveClass('custom-testimonial')
```

Comment:
Could we also ensure custom classes don't overwrite the default classes please? One of the general edge case tests that should apply to all components, as it's led to bug reports in the past.

Example:

```suggestion
    const el = getByRole('figure');
    expect(el).toHaveClass('custom-testimonial')
    expect(el).toHaveClass('Testimonial')'
```

Also just to note that for readability and style preference, can we extract the element matcher outside of assertion please